[[ Bug 21916 ]] Fix memory leak when using printing on macOS

runrevmark opened the pull request:
This patch fixes a leak of a PMPaper object when the macOS printer
is reconfigured/reset. The leak is caused by failing to release the
previous held PMPaper object before it is updated with a revised
one when setting the system printer properties.

livecodeian commented on engine/src/osxprinter.cpp:
```diff
@@ -447,8 +447,6 @@ bool MCMacOSXPrinter::Reset(MCStringRef p_name, PMPrintSettings p_settings, PMPa
 			t_success = false;
-		else
-			PMRetain(t_paper);
 	}
```
I'm not sure this is correct - according to the docs here https://developer.apple.com/documentation/applicationservices/1461319-pmgetpageformatpaper?language=objc the returned PMPaper value should not be released without retaining first.

runrevmark replied:
Okay - I'll take another look - it definitely fixed the leak I saw, but perhaps not in the right way :)

runrevmark committed:
[[ Bug 21916 ]] Fix memory leak when using printing on macOS

This patch fixes a leak of a PMPaper object when the macOS printer
is reconfigured/reset. The leak is caused by failing to release the
previous held PMPaper object before it is updated with a revised
one when setting the system printer properties.

runrevmark commented:
@livecodeali / @livecodeian - So the original fix was wrong, the actual leak was occurring due to a failure to release the previously set PMPaper, rather than an over-retain on reset :)

livecodeian commented:
@livecode-vulcan review ok e7c138a

livecode-vulcan commented:
💙 review by @livecodeian ok e7c138a8c143c333c9ca0949dbfff550217bbd13

livecode-vulcan commented:
😎 test success e7c138a8c143c333c9ca0949dbfff550217bbd13

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/460)
* try-community-armv7-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv7-android-sdk26_ndk16r15/builds/452)
* try-community-arm64-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-arm64-android-sdk26_ndk16r15/builds/452)
* try-community-x86-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86-android-sdk26_ndk16r15/builds/452)
* try-community-x86_64-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86_64-android-sdk26_ndk16r15/builds/452)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/1974)
* try-community-universal-ios-iphoneos12.1: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos12.1/builds/202)
* try-community-universal-ios-iphonesimulator12.1: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator12.1/builds/201)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/2005)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/1996)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/1994)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/2022)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/1963)
